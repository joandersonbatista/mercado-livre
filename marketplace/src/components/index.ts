export { Header } from './header/header';
export { IconButton } from './icon-button/icon-button';
export { Button } from './button/button';
export { Breadcrumbs } from './breadcrumbs/breadcrumbs';
export { BreadcrumbsSkeleton } from './breadcrumbs/breadcrumbs-skeleton';
export { Skeleton, type SkeletonProps } from './skeleton/skeleton';
export { LanguageSwitcher } from './language-switcher/language-switcher';
export { ServerError } from './server-error/server-error';
