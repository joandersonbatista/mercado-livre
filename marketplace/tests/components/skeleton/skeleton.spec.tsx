import * as React from 'react';

import { Skeleton } from '@/components';
import { render } from '@testing-library/react';

describe('Skeleton Component', () => {
  it('should render with default width and height', () => {
    const { getByTestId } = render(<Skeleton />);
    const divElement = getByTestId('skeleton');

    expect(divElement).toHaveStyle('width: undefined');
    expect(divElement).toHaveStyle('height: undefined');
  });

  it('should apply custom width and height when provided', () => {
    const { getByTestId } = render(<Skeleton width="100px" height="50px" />);
    const divElement = getByTestId('skeleton');

    expect(divElement).toHaveStyle('width: 100px');
    expect(divElement).toHaveStyle('height: 50px');
  });

  it('should render with the skeleton class', () => {
    const { getByTestId } = render(<Skeleton />);
    const divElement = getByTestId('skeleton');

    expect(divElement).toHaveClass('skeleton');
  });

  it('should match the snapshot', () => {
    const { container } = render(<Skeleton />);
    expect(container).toMatchSnapshot();
  });
});
