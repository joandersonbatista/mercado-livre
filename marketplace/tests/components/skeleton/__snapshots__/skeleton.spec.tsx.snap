// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component should match the snapshot 1`] = `
<div>
  <div
    class="skeleton"
    data-testid="skeleton"
  />
</div>
`;
